Added Obsidian Mathlive to community-plugins.json

danzilberdan opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

danzilberdan commented:
Hi :) Can someone check the plugin and help me get it to more people? Thanks!

joethei commented:
[interface MyPluginSettings {](https://github.com/danzilberdan/obsidian-mathlive/blob/a5e6b2c0e7078883749634ec5a3af6a40b8bf212/main.tsx#L7-L13)
Since you don't use them you can remove all references to settings.

[id: 'open-mathlive-modal-line',](https://github.com/danzilberdan/obsidian-mathlive/blob/a5e6b2c0e7078883749634ec5a3af6a40b8bf212/main.tsx#L22)
Obsidian will automatically prefix the command id with the plugin id,
you don't need to do include it yourself.

[console.log("The parsing result is", parseResult);](https://github.com/danzilberdan/obsidian-mathlive/blob/a5e6b2c0e7078883749634ec5a3af6a40b8bf212/main.tsx#L104)
[console.log("Running onClose", this.renderedResult)](https://github.com/danzilberdan/obsidian-mathlive/blob/a5e6b2c0e7078883749634ec5a3af6a40b8bf212/main.tsx#L126)
[console.log("Rendering MathLive into the the div.");](https://github.com/danzilberdan/obsidian-mathlive/blob/a5e6b2c0e7078883749634ec5a3af6a40b8bf212/MathLiveComponent.tsx#L9)
Remove this.

danzilberdan commented:
Hi! I have fixed all of the requests.
Thanks for the help :)

danzilberdan commented:
Hi Can you checkout the updated branch? Thanks

liamcain commented:
- [MyPlugin](https://github.com/danzilberdan/obsidian-mathlive/blob/86a33598bed3e6b50c7833f5ac5ff080f35f0e39/main.tsx#L7) you should give your class a meaningful name.
- [checkCallback](https://github.com/danzilberdan/obsidian-mathlive/blob/86a33598bed3e6b50c7833f5ac5ff080f35f0e39/main.tsx#L13) you should use [editorCallback](https://github.com/obsidianmd/obsidian-api/blob/38dd22168d2925086371bfc59e36fd9121527a39/obsidian.d.ts#L614) here instead. In your onClose function, you're assuming editor.replaceSelection() will work, but that will only work if the user is in Editing mode. changing checkCallback to editorCallback will automatically ensure the command only triggers in Editing mode.
- [constructor](https://github.com/danzilberdan/obsidian-mathlive/blob/86a33598bed3e6b50c7833f5ac5ff080f35f0e39/main.tsx#L33) I suggest passing "editor" into the MathLiveModal
- [MathLiveComponent](https://github.com/danzilberdan/obsidian-mathlive/blob/86a33598bed3e6b50c7833f5ac5ff080f35f0e39/MathLiveComponent.tsx#L5) I'm not really sure why you're using React at all for this plugin. You could skip the MathLiveComponent completely and it would only simplify your code as far as I can tell.

ericaxu commented:
Hi @danzilberdan , do you still intend to submit this plugin?

danzilberdan commented:
Yes. I will fix the issues.

danzilberdan commented:
Hi! I have fixed the issues.
1. Removed react.
2. Added a new button that will make inserting the formula more intuitive.

Sorry for the delay and Thanks.
